### PR TITLE
Update example to use new func RegisterWithErrorFunc

### PR DESCRIPTION
**Pull request by giulianopz**

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -42,11 +43,10 @@ func ExampleNfqueue_Register() {
 	}
 
 	// Register your function to listen on nflqueue queue 100
-	err = nf.Register(ctx, fn)
```
Comment (florianl) [inline example_test.go:45]: Replacing the deprecated function `Register()` with `RegisterWithErrorFunc()` in the example is a good idea. But we should keep the error handling.

**Review by florianl — COMMENTED** [review]
Thanks for your contribution. 

Could you bring back the error handling and sign your commit, please?

**Comment by giulianopz** [comment]
done!

File: example_test.go [hunk 2]
```diff
@@ -42,11 +43,13 @@ func ExampleNfqueue_Register() {
...
-		fmt.Println(err)
-		return
-	}
+	err = nf.RegisterWithErrorFunc(ctx, fn, func(e error) int {
+		if err != nil {
+			fmt.Println(err)
+			return -1
+		}
+		return 0
+	})
```
Comment (florianl) [inline example_test.go:52]: If [ErrorFunc](https://pkg.go.dev/github.com/florianl/go-nfqueue#ErrorFunc) is called in the scope of `RegisterWithErrorFunc()`, then error is never nil.
But `RegisterWithErrorFunc()` also returns an error, e.g. if the hook can not be registered to the socket. Then this error should not be ignored.
```suggestion
	err = nf.RegisterWithErrorFunc(ctx, fn, func(e error) int {
		fmt.Println(err)
		return -1
	})
	if err != nil {
		fmt.Println(err)
		return
	}
```

Comment (giulianopz) [reply example_test.go]: Sorry, I misunderstood your original request. Changed as suggested.

Comment (florianl) [reply example_test.go]: Thanks for the update - it was my mistake not to be precious about my requested changes.

**Review by florianl — APPROVED** [review]

**Pull request merged**